traffic_ops/traffic_ops_golang/routes.go

```diff
@@ -407,6 +408,9 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.1, http.MethodGet, `deliveryservices/{id}/urlkeys/?(\.json)?$`, deliveryservice.GetURLKeysByID, auth.PrivLevelReadOnly, Authenticated, nil},
 		{1.1, http.MethodGet, `riak/bucket/{bucket}/key/{key}/values/?(\.json)?$`, apiriak.GetBucketKey, auth.PrivLevelAdmin, Authenticated, nil},
 
+		//Pattern based consistent hashing endpoint
+		{1.1, http.MethodPost, `consistenthash/?$`, consistenthash.Post, auth.PrivLevelReadOnly, Authenticated, nil},
```

Comment:
i believe this should be 1.4...or maybe even 1.5...

@rawlinp @rob05c

Reply:
Yeah, Release 3.0.0 has 1.4, so this needs to be 1.5.

Reply:
In that case, the DS structs need to be re-done - iirc this PR would add a field to them starting at 1.3

Reply:
The structs are easy change, but right now, the endpoint itself needs a lot of copy-paste. It didn't used to be too bad, but it got considerably worse with all the "CRUDer" boilerplate.

The Go language has very limited metaprogramming, which is what we need here. So, there's no ideal solution. But, I'm really hoping to come up with something better than what we have. Or if not, I'm willing to make the new endpoint, rather than requiring Jesse to. As long as we fix it before the next Release, it's ok for `master` to be imperfect between releases.

So, I'm inclined to Approve this, as long as we make a Github issue to fix it before the next release. If you're ok with that @ocket8888 ?

Reply:
Sure, but it'll mean that something documented as part of v1.3 will later be revised to be part of v1.5 (god I hope the issue resolution doesn't miss that) or we could straight-up lie in the documentation and say it's only available in v1.5, then link to the issue and call the field's appearance in v1.3 and v1.4 a "bug".

Reply:
To be clear, I'm fine with either, personally. I think the best solution is the latter, since even if the issue resolution misses cleaning that up, people can follow the issue link and see that it actually was resolved.